tutorials/physics/using_jolt_physics.rst

```diff
@@ -0,0 +1,504 @@
+JoltHingeJoint3D, JoltSliderJoint3D, JoltConeTwistJoint3D and JoltGeneric6DOFJoint)
+have not been included with this module.
+
+Thread Safety
+~~~~~~~~~~~~~
+
+Unlike the Godot Jolt extension, this module does have experimental thread-safety,
+including support for the :ref:`physics/3d/run_on_separate_thread<class_ProjectSettings_Property_physics/3d/run_on_separate_thread>`
+project setting. This is achieved by utilizing the same wrapper server that's used
+by Godot Physics, called ``PhysicsServer3DWrapMT``, as well as introducing a mutex
+around the rebuilding of shapes, since concurrent shape-casts could otherwise
+trigger a race condition.
+
+This has however not been tested very thoroughly, so should be considered
+experimental.
+
+Query Performance
+~~~~~~~~~~~~~~~~~
+
+The Godot Jolt extension utilizes a custom container (typically referred to as an
+"inline vector" or "small vector") in order to avoid heap allocations for physics
+queries that return more than a single hit, meaning :ref:`intersect_point<class_PhysicsDirectSpaceState3D_method_intersect_point>`,
+:ref:`intersect_shape<class_PhysicsDirectSpaceState3D_method_intersect_shape>`,
+:ref:`collide_shape<class_PhysicsDirectSpaceState3D_method_collide_shape>`,
+:ref:`get_rest_info<class_PhysicsDirectSpaceState3D_method_get_rest_info>`,
+:ref:`cast_motion<class_PhysicsDirectSpaceState3D_method_cast_motion>`,
+:ref:`body_test_motion<class_PhysicsServer3D_method_body_test_motion>`,
+:ref:`test_move<class_PhysicsBody3D_method_test_move>`, :ref:`move_and_collide<class_PhysicsBody3D_method_move_and_collide>`
+and :ref:`move_and_slide<class_CharacterBody3D_method_move_and_slide>`.
+
+For the built-in module ``JPH::Array`` is used for the query collectors instead, which
+means that these queries will now always allocate on the heap, likely making them
+noticeably slower.
+
+It should be trivial to make some bespoke container for the query collectors that
+behaves like an inline/small vector.
+
+Debug Renderer
+~~~~~~~~~~~~~~
+
+Jolt provides an interface called ``JPH::DebugRenderer`` for rendering its view of
+the physics simulation. In the Godot Jolt extension this is exposed as a custom
+:ref:`class_GeometryInstance3D` node called ``JoltDebugGeometry3D``, a different
+approach will be used for the built-in module, and for now it is omitted.
+
+Debug snapshots
+~~~~~~~~~~~~~~~
+
+The Godot Jolt extension has the ability to generate what Jolt refers to as
+"snapshots", where it serializes the state of the physics simulation to a file.
+These can then be loaded in Jolt's own Samples application, to debug issues more
+closely there, without needing to deal with Godot itself, which has proved to be
+quite useful when reporting issues upstream to Jolt.
+
+The code for this is technically included with this module, and can be found as
+``JoltPhysicsServer3D::dump_debug_snapshots``, but it is currently unexposed for now.
+
+Roadmap
```

Comment:
I would strike this whole section, it feels out of place with other docs - the docs are generally not a place to talk about future plans, and we don't use it for roadmaps for other teams AFAIK.

The whole future being experimental is fair, but IMO some other place (a blog post? the original PR?) are a better time capsule to capture the current state of what is in, what is going to be in, and what won't be in than the docs.